x/authz/types/msgs.go

```diff
@@ -0,0 +1,247 @@
+package types
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"gopkg.in/yaml.v2"
+
+	types "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// authz message types
+const (
+	TypeMsgGrantAuthorization  = "grant_authorization"
+	TypeMsgRevokeAuthorization = "revoke_authorization"
+	TypeMsgExecDelegated       = "exec_delegated"
+)
+
+var (
+	_ sdk.Msg = &MsgGrantAuthorization{}
+	_ sdk.Msg = &MsgRevokeAuthorization{}
+	_ sdk.Msg = &MsgExecAuthorized{}
```

Comment:
```suggestion
	_ sdk.RequestMsg = &MsgGrantAuthorization{}
	_ sdk.RequestMsg = &MsgRevokeAuthorization{}
	_ sdk.RequestMsg = &MsgExecAuthorized{}
```

We don't need to support all the legacy `sdk.Msg` methods, just` sdk.RequestMsg`.